CMakeLists.txt

```
@@ -203,11 +212,14 @@ endif()
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
   # For some reason, adding to CMAKE_CXX_FLAGS results in a slightly smaller
   # binary than using `add_compile/link_options`
-  set(_extra_flags "-ffunction-sections -fdata-sections -Wl,--gc-sections")
+  set(_extra_flags "-ffunction-sections -fdata-sections -Wl,--gc-sections -Wl,--as-needed")

```

Shouldn't this also work?
```suggestion
  set(_extra_flags "-ffunction-sections -fdata-sections -Wl,--gc-sections,--as-needed")
```

---

Probably, I'm not near a computer to try but feel free to make this change if it works! 👍

---

Done